### PR TITLE
Pybind test 

### PR DESCRIPTION
**Pull request by akawashiro**
This PR tests that `sold` can handle pybind-related stuff including NumPy bindings. In addition to it, I changed we can run all tests with `ctest` command.

### CODE REVIEW

File: sold.cc [hunk 1]
```diff
@@ -384,8 +384,8 @@ uintptr_t Sold::TLSMemSize() const {
 void Sold::DecideMemOffset() {
     uintptr_t offset = 0x10000000;
     for (ELFBinary* bin : link_binaries_) {
-        const Range range = bin->GetRange() + offset;
-        CHECK(range.start == offset);
+        const Range range = bin->GetRange() + offset - bin->GetRange().start;
```
Comment (shinh) [inline sold.cc:387]: Could you tell me which .so file has `bin->GetRange().start != 0`? I'm not sure if subtracting `start` is the right fix for it.  Have you considered a fix like

```
-        CHECK(range.start == offset);
-        offsets_.emplace(bin, range.start);
+        offsets_.emplace(bin, offset);
```

I'm not sure at all this is better. However, this makes a bit more sense to me than subtracting `start`.

Comment (akawashiro) [reply sold.cc:387]: `bin->GetRange().start == 0` is true for executables.

Comment (akawashiro) [reply sold.cc]: As I understood the intention of the code, I restore it and added a comment. https://github.com/shinh/sold/pull/75/commits/52807b61d5cfe99c990380b4e48e56a29cb9a887

**Review by shinh — APPROVED** [review]
LGTM, thanks!

**Pull request merged**